airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/InitialPkLoadEnabledCdcMysqlSourceTest.java

```diff
@@ -0,0 +1,355 @@
+
+      stateMessage.getGlobal().getStreamStates().forEach(s -> {
+        final JsonNode streamState = s.getStreamState();
+        if (s.getStreamDescriptor().equals(new StreamDescriptor().withName(MODELS_STREAM_NAME + "_random").withNamespace(randomTableSchema()))) {
+          assertEquals(PRIMARY_KEY_STATE_TYPE, streamState.get(STATE_TYPE_KEY).asText());
+        } else if (s.getStreamDescriptor().equals(new StreamDescriptor().withName(MODELS_STREAM_NAME).withNamespace(MODELS_SCHEMA))) {
+          assertFalse(streamState.has(STATE_TYPE_KEY));
+        } else {
+          throw new RuntimeException("Unknown stream");
+        }
+      });
+    }
+
+    final AirbyteStateMessage secondLastSateMessage = stateMessages.get(5);
+    assertEquals(AirbyteStateMessage.AirbyteStateType.GLOBAL, secondLastSateMessage.getType());
+    assertEquals(stateMessageEmittedAfterFirstSyncCompletion.getGlobal().getSharedState(),
+        secondLastSateMessage.getGlobal().getSharedState());
+    final Set<StreamDescriptor> streamsInSnapshotState = secondLastSateMessage.getGlobal().getStreamStates()
+        .stream()
+        .map(AirbyteStreamState::getStreamDescriptor)
+        .collect(Collectors.toSet());
+    assertEquals(2, streamsInSnapshotState.size());
+    assertTrue(
+        streamsInSnapshotState.contains(new StreamDescriptor().withName(MODELS_STREAM_NAME + "_random").withNamespace(randomTableSchema())));
+    assertTrue(streamsInSnapshotState.contains(new StreamDescriptor().withName(MODELS_STREAM_NAME).withNamespace(MODELS_SCHEMA)));
+    secondLastSateMessage.getGlobal().getStreamStates().forEach(s -> {
+      final JsonNode streamState = s.getStreamState();
+      assertFalse(streamState.has(STATE_TYPE_KEY));
+    });
+
+    final AirbyteStateMessage stateMessageEmittedAfterSecondSyncCompletion = stateMessages.get(6);
+    assertEquals(AirbyteStateMessage.AirbyteStateType.GLOBAL, stateMessageEmittedAfterSecondSyncCompletion.getType());
+    assertNotEquals(stateMessageEmittedAfterFirstSyncCompletion.getGlobal().getSharedState(),
+        stateMessageEmittedAfterSecondSyncCompletion.getGlobal().getSharedState());
+    final Set<StreamDescriptor> streamsInSyncCompletionState = stateMessageEmittedAfterSecondSyncCompletion.getGlobal().getStreamStates()
+        .stream()
+        .map(AirbyteStreamState::getStreamDescriptor)
+        .collect(Collectors.toSet());
+    assertEquals(2, streamsInSnapshotState.size());
+    assertTrue(
+        streamsInSyncCompletionState.contains(
+            new StreamDescriptor().withName(MODELS_STREAM_NAME + "_random").withNamespace(randomTableSchema())));
+    assertTrue(streamsInSyncCompletionState.contains(new StreamDescriptor().withName(MODELS_STREAM_NAME).withNamespace(MODELS_SCHEMA)));
+    assertNotNull(stateMessageEmittedAfterSecondSyncCompletion.getData());
+  }
+
+  @Override
+  @Test
+  protected void syncShouldHandlePurgedLogsGracefully() throws Exception {
```

Comment:
This test should not work right? Cause right now we dont have the mechanism to check whether the state that we have is purged from the server or not in your branch

Reply:
I have this comment on line 192, which currently asserts the old DBZ snapshot codepath is taken (and not PK load). Once we have that mechanism to check binlog expiry, this test will need to be updated  
    // Purge the binary logs. The current code reverts to the debezium snapshot when binary logs are purged, and does
    // not do an initial primary key load. Thus, we only expect one state message for now.